File: android/app/src/main/java/com/clingsync/android/data/UploadStatus.kt
package com.clingsync.android.data

// The status vocabulary the upload worker writes to its progress/result files.
// Centralised here so the worker (writer) and the UI (reader) share one schema.
enum class UploadStatus(val wire: String) {
    WAITING("waiting"),
    UPLOADING("uploading"),
    UPLOADED("uploaded"),
    SKIPPED("skipped"),
    COMMITTING("committing"),
    ;

    companion object {
        fun fromWire(wire: String): UploadStatus? = entries.firstOrNull { it.wire == wire }
    }
}

File: ios/ClingSync/UploadStatus.swift
import Foundation

// The status vocabulary the upload mechanism writes to its progress/result files.
// Centralised here so the writer (the upload coordinator) and the reader (the
// store) share one schema. The raw value IS the on-disk wire string.
enum UploadStatus: String {
    case waiting
    case uploading
    case uploaded
    case skipped
    case committing

    var wire: String { rawValue }

    static func fromWire(_ wire: String) -> UploadStatus? { UploadStatus(rawValue: wire) }
}
